src/import/import-generator.c

```diff
@@ -227,24 +259,39 @@ static int transfer_generate(sd_json_variant *v, size_t c) {
                 fprintf(f, "FailureAction=%s\n",
                         arg_failure_action);
 
-        const char *class = sd_json_variant_string(sd_json_variant_by_key(v, "class"));
-        if (streq_ptr(class, "sysext"))
+        if (t->class == IMAGE_SYSEXT)
                 fputs("Before=systemd-sysext.service\n", f);
-        else if (streq_ptr(class, "confext"))
+        else if (t->class == IMAGE_CONFEXT)
                 fputs("Before=systemd-confext.service\n", f);
 
         /* Assume network resource unless URL is file:// */
-        if (!file_url_is_valid(remote))
+        if (!file_url_is_valid(t->remote))
                 fputs("Wants=network-online.target\n"
                       "After=network-online.target\n", f);
 
+        _cleanup_free_ char *local_path = NULL, *loop_service = NULL;
+        if (t->make_blockdev) {
+                assert(t->type == IMPORT_RAW);
+
+                local_path = strjoin(image_root_to_string(t->class), "/", t->local, ".raw");
```

Comment:
`t->local` may be NULL ??

I think it is better to generate local name from the remote URL if unspecified.

Reply:
we do that actually, but in importd, server side.